### DISCUSSION TITLE
Use of sha256 as digest algorithm

### DISCUSSION THREAD

**Discussion by Yash-Khatri** [rpm-software-management/rpm#2037] [Q&A] [answered]
Hi,

I am trying to use sha256 algo for digest for my RPM packages. currently, it is building with default sha1 and md5 digests.

Here is what I tried so far

- Upgraded the RPM to version 4.14.3. which supports the sha256 as default algo for digests.

- Override macros for the digest algorithm by adding the below lines in **$HOME/.rpmmacros file or /etc/rpm/macros** file [reference](https://www.starlab.io/blog/adding-sha256-digests-to-rpms)
```
%_source_filedigest_algorithm 8
%_binary_filedigest_algorithm 8
```
- Add the above macros in pkg.spec file as below which is being passed to rpmbuild command
```
%define _source_filedigest_algorithm 8
%define _binary_filedigest_algorithm 8
```
but none of the above solutions worked for me and after rebuilding I still see my package is building with sha1 and md5 as digest algo.

Any help is appreciated.

**Comment by pmatilai** [comment, marked as answer]
Those macros control the *per file* digest algorithm that is used for verifying unpacked files and rpm -V. 
The *per package* digests are not configurable, RPMTAG_SHA256HEADER and RPMTAG_PAYLOADDIGEST (which is SHA256) are only present in packages built with rpm >= 4.14.

**Reply by Yash-Khatri** [reply]
Thanks for the clarification @pmatilai.

Is there any way we can upgrade the rpm version to >=4.14 on the **rhel6** or **rhel7** systems, where the default rpm version is 4.8 and 4.11 respectively?

Without building the source itself on my own, from where can I get the official rpm packages?

**Reply by pmatilai** [reply]
It doesn't work like that. Like @ffesti said in #2038 , you should build packages with the distro and it's rpm you're targetting.
